web/static/js/components/thesis-editor.js

```diff
@@ -170,6 +178,29 @@ class ThesisEditor extends React.Component {
     })
   }
 
+  importData (page, contents) {
+    // apply page settigns
+    this.setState({
+      importContentQueueCount: contents.length,
+      importContentCompletedCount: 0,
+      title: page.title,
+      description: page.description,
+      pageModified: true
+    })
+
+    // apply page contents
+    for (let i = 0; i < contents.length; i++) {
+      const content = contents[i]
+      this.importContent(content.content_type, content, page)
+    }
+
+    window.importContentCounterTimeout = setTimeout(() => {
+      this.setState({
+        importProgress: {type: 'error', 'text': 'Something went wrong! Please refresh and try again.'}
+      })
+    }, 20000)
```

Comment:
at this point, the import/export tray is still open and telling the user that the data is being imported. this timeout is a just in case something goes wrong, to notify the user in 20 seconds to try again. 

normally, the data should propagate immediately, but if there are a lot of images, it make take some time. i think 20 seconds is plenty though even for very media heavy pages. we can test though.